File: src/main/java/com/project/app/model/Rate.java
package com.project.app.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.time.LocalDate;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity(name = "rates")
public class Rate {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    private Currency currency;

    @NotNull
    private Double value;

    @NotNull
    private LocalDate date;
}

File: frontend/src/app/model/rate.ts
import {Currency} from "./currency";

export class Rate {
  id: number;
  currency: Currency;
  value: number;
  lastChange: number;
  date: string;

  constructor(currency: Currency, value: number, lastChange: number, date: string) {
    this.currency = currency;
    this.value = value;
    this.lastChange = lastChange;
    this.date = date;
  }
}
